pkg/runtime/runtime.go

```diff
@@ -1642,18 +1645,18 @@ func (a *DaprRuntime) publishMessageGRPC(ctx context.Context, msg *pubsubSubscri
 	}
 
 	var span *trace.Span
-	if iTraceID, ok := cloudEvent[pubsub.TraceIDField]; ok {
-		if traceID, ok := iTraceID.(string); ok {
-			sc, _ := diag.SpanContextFromW3CString(traceID)
-			spanName := fmt.Sprintf("pubsub/%s", msg.topic)
-
-			// no ops if trace is off
-			ctx, span = diag.StartInternalCallbackSpan(ctx, spanName, sc, a.globalConfig.Spec.TracingSpec)
-			ctx = diag.SpanContextToGRPCMetadata(ctx, span.SpanContext())
-		} else {
-			log.Warnf("ignored non-string traceid value: %v", iTraceID)
-		}
-	}
+	traceParent, _ := cloudEvent[pubsub.TraceParentField].(string)
```

Comment:
In order to preserve backwards compat, we need to check for either trace parent or trace ID.

Reply:
This method name is a bit confusing - this is Dapr sending the message to the consumer.